src/thermo/ThermoFactory.cpp

```diff
@@ -129,10 +129,32 @@ ThermoPhase* newPhase(const std::string& infile, std::string id)
 
     if (extension == "yml" || extension == "yaml") {
         AnyMap root = AnyMap::fromYamlFile(infile);
-        AnyMap& phase = root["phases"].getMapWhere("name", id);
-        unique_ptr<ThermoPhase> t(newThermoPhase(phase["thermo"].asString()));
-        setupPhase(*t, phase, root);
-        return t.release();
+        std::string name = id;
+        try {
+            AnyMap& phase = root["phases"].getMapWhere("name", name);
+            unique_ptr<ThermoPhase> t(newThermoPhase(phase["thermo"].asString()));
+            setupPhase(*t, phase, root);
+            return t.release();
+        } catch (CanteraError& err) {
+            if (name=="gri30_mix" || name=="gri30_multi") {
+                warn_deprecated("newPhase",
+                    "\nThe phases named 'gri30_mix' and 'gri30_multi' are no "
+                    "longer defined in the input file 'gri30.yaml'. By default, "
+                    "'gri30.yaml' uses the mixture-averaged transport model. The "
+                    "following options are available:\n"
+                    " * 'Mix': Mixture-averaged transport\n"
+                    " * 'Multi': Multicomponent transport\n"
+                    " * 'None': No transport model, useful for 0-D simulations\n"
```

Comment:
It would be more useful to show how users can fix this warning - where does this option get passed?

Reply:
Will add a line.

Reply:
Added examples for MATLAB and C++